mysensors/mysensors.py

```diff
@@ -280,14 +297,15 @@ def _perform_file_action(self, filename, action):
             raise Exception('Unsupported file type %s' % ext[1:])
         func(filename)
 
-    def alert(self, nid):
+    def alert(self, msg):
         """Tell anyone who wants to know that a sensor was updated.
 
         Also save sensors if persistence is enabled.
         """
         if self.event_callback is not None:
             try:
-                self.event_callback('sensor_update', nid)
+                setattr(msg, 'gateway', self)
```

Comment:
I can see why this would be convenient, but I think it's confusing to change the structure of the message instance here. If the user of the library want to take advantage of the info in the message, the user will need to know the structure of the `Message` class. But the gateway is not an attribute of the `Message` class, so that makes the addition here confusing. Also I think the user would always already have access to the gateway instance, after instantiating one of the gateway classes.